framework/src/main/java/org/checkerframework/framework/flow/CFAbstractValue.java

```diff
@@ -292,19 +290,12 @@ private class MostSpecificVisitor extends AnnotationSetCombiner {
         /**
          * Create a {@link MostSpecificVisitor}.
          *
-         * @param aTypeMirror type of the "a" value
-         * @param bTypeMirror type of the "b" value
          * @param backup value to use if no most specific value is found
          */
-        @SuppressWarnings("UnusedVariable") // TODO clean this up
-        public MostSpecificVisitor(TypeMirror aTypeMirror, TypeMirror bTypeMirror, V backup) {
+        public MostSpecificVisitor(V backup) {
```

Comment:
Thanks for the cleanup!
Above, `if (ms.error) { return backup; }`, is `backup` always null if `ms.error` is true?
A few nullness annotations also seem to be missing, e.g. on `backupAMSet`.

Reply:
>is backup always null if ms.error is true?
I think no, the visitor will not reassign the value of `backup`. It only depends on the input.

>A few nullness annotations also seem to be missing, e.g. on backupAMSet.
I will add it.

Reply:
> > is backup always null if ms.error is true?
> 
> I think no, the visitor will not reassign the value of `backup`. It only depends on the input.

I don't understand what you mean.
The only place where `error` is set to `true` is when `if (backupAMSet == null)`.
I think the only way that can happen is if `backup` is null.
So in

````
        if (ms.error) {
            return backup;
        }
````

I would expect that the returned value is always null.
Wouldn't the code be clearer if `null` is returned explicitly, possibly with a comment that `backup == null`?
If that argument is flawed, what am I missing?

Reply:
Thanks for clarification. See new commit [d2baafd](https://github.com/eisop/checker-framework/pull/761/commits/d2baafddf07d604b9e427be99cd0fc6d61912ab0)